src/Datalust.SerilogMiddlewareExample/Diagnostics/SerilogMiddleware.cs

```diff
@@ -65,9 +66,10 @@ static ILogger LogForErrorContext(HttpContext httpContext)
             return result;
         }
 
-        static double GetElapsedMilliseconds(long start, long stop)
+        private static PathString GetPath(HttpContext httpContext)
         {
-            return (stop - start) * 1000 / (double)Stopwatch.Frequency;
+            var rawTarget = httpContext.Features.Get<IHttpRequestFeature>()?.RawTarget;
```

Comment:
If we expect the `IHttpRequestFeature` to usually or mostly be present, perhaps we could return `string` from this method to avoid allocating (and initializing - I think it does some parsing) a `PathString`? The old branch code would then be `httpContext.Request.Path.ToString()`, which will be called by the logging logic anyway.

Reply:
Agree on the performance hit, my concern was writing RawTarget to the logs without parsing it first.  Can you think of any security issues of writing it? The only one I can think of would be if large strings aren't handled properly.

Reply:
Worth considering, but I think this will be fine, the HTTP server and ASP.NET Core have both already accepted the request by this point - I'd be surprised if a gigabyte raw URL would get this far.

Reply:
Good point, I can't come up with any other security issues so I've updated to just return the strings without PathString. 

BTW, I'm using a modified version of it in a 2.1 template I've started working on at: https://github.com/jmhmaine/mvc-opinionated-template/blob/master/src/MvcOpinionatedTemplate.Web/SerilogMiddleware.cs 

Besides a some syntax differences, The only significant functional difference is that I'm suppressing writing the 500 error in the InvokeAsync try. The reason is that the meaningful information is already logged in your catch when logging statement.  

The only thing I need to research is if there are any unique 500 errors that would appear in the try that wouldn't be thrown and already caught. 

I'm also overriding the min level up to Fatal to reduce duplicate log events:
   .MinimumLevel.Override("Microsoft", LogEventLevel.Fatal)
   .MinimumLevel.Override("System", LogEventLevel.Fatal)

I need to also research if there is anything not being caught by Serilog with these overrides.